### DISCUSSION TITLE
error when running a Rosetta tutorial on Thermostabilization and general protein design

### DISCUSSION THREAD

**Discussion by biocatchen** [RosettaCommons/rosetta#163] [Rosetta]
Hi all,
I used MacOS M1 system to run this Rosetta protocol (https://meilerlab.org/wp-content/uploads/2022/02/May2018_thermostabilization_tutorial.pdf). It ended with an error when running Part II.2: Transform the proteins into the membrane and evaluate the outward-facing and inward-facing conformations of wildtype LeuT.
I noticed that the tutorial protocol uses linux system, therefore the script score_jd2.linuxgccrelease. I replaced it with macOS-based script i.e. score_jd2.static.macosclangrelease under rosetta 3.14_arm version (I cannot find the exact equivalent script like score_jd2.macosclangrelease).

Here is the output containing error message below. Any suggestions are welcome!
I also attach the crash.log file for your reference!
[ROSETTA_CRASH.log](https://github.com/user-attachments/files/17007097/ROSETTA_CRASH.log)

********  (C) Copyright Rosetta Commons Member Institutions.  ***************
* Use of Rosetta for commercial purposes may require purchase of a license. *
********  See LICENSE.md or email license@uw.edu for more details. **********
core.init: Checking for fconfig files in pwd and ./rosetta/flags 
core.init: Rosetta version: rosetta.binary.m1.release-371 r371 2024.09+release.06b3cf8ad0 06b3cf8ad0940d628690d0ed6fa2009d72ad2b44 https://www.rosettacommons.org 2024-02-28T21:25:09.004217
core.init: command: /opt/sbgrid/i386-mac/rosetta/3.14_arm/main/source/bin/score_jd2.static.macosclangrelease @score.options -in:file:s 2A65_full.pdb 3TT3_full.pdb -out:pdb
basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1203533217 seed_offset=0 real_seed=1203533217
basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1203533217 RG_type=mt19937
core.init: Resolved executable path: /opt/sbgrid/i386-mac/rosetta/3.14_arm/main/source/bin/score_jd2.static.macosclangrelease
core.init: Looking for database based on location of executable: /opt/sbgrid/i386-mac/rosetta/3.14_arm/main/database/
core.scoring.ScoreFunctionFactory: SCOREFUNCTION: mpframework_smooth_fa_2012.wts
core.scoring.etable: Starting energy table calculation
core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well
core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)
core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)
core.scoring.etable: Finished calculating energy tables.
basic.io.database: Database file opened: scoring/score_functions/PairEPotential/pdb_pair_stats_fine
core.scoring.etable: Starting energy table calculation
core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well
core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)
core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)
core.scoring.etable: Finished calculating energy tables.
core.scoring.etable: Using alternate parameters: LK_DGFREE in MembEtable construction.
core.scoring.etable: Using alternate parameters: MEMB_LK_DGFREE in MembEtable construction.
core.scoring.etable: Using alternate parameters: LK_DGREFCE in MembEtable construction.
core.scoring.etable: Using alternate parameters: MEMB_LK_DGREFCE in MembEtable construction.
core.scoring.etable: Starting membrane specific energy table calculation
basic.io.database: Database file opened: scoring/score_functions/MembranePotential/CEN10_Menv_smooth_log.txt
basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv
basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv
basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv
basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv
basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv
basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA
basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n
core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.
basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop
core.scoring.ramachandran: shapovalov_lib::shap_rama_smooth_level of 4( aka highest_smooth ) got activated.
basic.io.database: Database file opened: scoring/score_functions/rama/shapovalov/kappa25/all.ramaProb
basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt
basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt
basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt
basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt
protocols.evaluation.ChiWellRmsdEvaluatorCreator: Evaluation Creator active ... 
protocols.jd2.PDBJobInputter: Instantiate PDBJobInputter
protocols.jd2.PDBJobInputter: PDBJobInputter::fill_jobs
protocols.jd2.PDBJobInputter: pushed 2A65_full.pdb nstruct index 1
protocols.jd2.PDBJobInputter: pushed 3TT3_full.pdb nstruct index 1
protocols.evaluation.ChiWellRmsdEvaluatorCreator: Evaluation Creator active ... 
protocols.evaluation.ChiWellRmsdEvaluatorCreator: Evaluation Creator active ... 
protocols.jd2.PDBJobInputter: PDBJobInputter::pose_from_job
protocols.jd2.PDBJobInputter: filling pose from PDB 2A65_full.pdb
core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types
core.chemical.GlobalResidueTypeSet: Total time to initialize 0.364941 seconds.
core.import_pose.import_pose: File '2A65_full.pdb' automatically determined to be of type PDB
protocols.membrane.AddMembraneMover: =====================================================================
protocols.membrane.AddMembraneMover: ||           WELCOME TO THE WORLD OF MEMBRANE PROTEINS...          ||
protocols.membrane.AddMembraneMover: =====================================================================
protocols.membrane.AddMembraneMover: No membrane residue was found
protocols.membrane.AddMembraneMover: Adding a new membrane residue to the pose
protocols.membrane.AddMembraneMover: Adding a membrane residue representing the position of the membrane after residue 519
core.chemical.GlobalResidueTypeSet: Loading (but possibly not actually using) 'MEM' from the PDB components dictionary for residue type 'pdb_MEM'
protocols.membrane.AddMembraneMover:    Edge   	   Jump     Jump #
          	0520--0001  001
0001--0519

core.conformation.membrane.SpanningTopology: Filling membrane spanning topology from spanfile LeuT.span
core.conformation.membrane.SpanningTopology: nres: 520
core.conformation.membrane.SpanningTopology: nres_topo: 519
core.conformation.membrane.SpanningTopology: total residues in spanfile: 519
core.conformation.membrane.SpanningTopology: Pose is in pose numbering scheme.
core.conformation.membrane.Span: [ WARNING ] LONG SPAN: SPAN IS 31 RESIDUES LONG!!!
core.conformation.membrane.Span: [ WARNING ] LONG SPAN: SPAN IS 31 RESIDUES LONG!!!
core.conformation.membrane.SpanningTopology: Total # of TM spans: 12
core.conformation.membrane.SpanningTopology: Number of residues in spanfile: 519
core.conformation.membrane.SpanningTopology: Span 1: start: 12, end: 32 orientation: 1
core.conformation.membrane.SpanningTopology: Span 2: start: 41, end: 61 orientation: 1
core.conformation.membrane.SpanningTopology: Span 3: start: 92, end: 122 orientation: 1
core.conformation.membrane.SpanningTopology: Span 4: start: 165, end: 185 orientation: 1
core.conformation.membrane.SpanningTopology: Span 5: start: 197, end: 217 orientation: 1
core.conformation.membrane.SpanningTopology: Span 6: start: 244, end: 264 orientation: 1
core.conformation.membrane.SpanningTopology: Span 7: start: 283, end: 303 orientation: 1
core.conformation.membrane.SpanningTopology: Span 8: start: 334, end: 364 orientation: 1
core.conformation.membrane.SpanningTopology: Span 9: start: 377, end: 397 orientation: 1
core.conformation.membrane.SpanningTopology: Span 10: start: 406, end: 426 orientation: 1
core.conformation.membrane.SpanningTopology: Span 11: start: 450, end: 470 orientation: 1
core.conformation.membrane.SpanningTopology: Span 12: start: 485, end: 505 orientation: 1
basic.io.database: Database file opened: membrane/implicit_lipid_parameters.txt
core.conformation.membrane.ImplicitLipidInfo: Reading electric field fitting parameters from the database
basic.io.database: Database file opened: membrane/lipid_electric_field_params.txt
core.conformation.membrane.ImplicitLipidInfo: lipid parameters are read
core.conformation.membrane.MembraneInfo: SLAB geometry
protocols.membrane.SetMembranePositionMover: Calling SetMembranePositionMover
protocols.membrane.SetMembranePositionMover: Starting foldtree: Is membrane fixed? 1
protocols.membrane.SetMembranePositionMover:    Edge   	   Jump     Jump #
          	0520--0001  001
0001--0519

protocols.membrane.SetMembranePositionMover: Final foldtree: Is membrane fixed? 1
protocols.membrane.SetMembranePositionMover:    Edge   	   Jump     Jump #
          	0520--0001  001
0001--0519

protocols.membrane.AddMembraneMover: Final foldtree: Is membrane fixed? 1
protocols.membrane.AddMembraneMover:    Edge   	   Jump     Jump #
          	0520--0001  001
0001--0519

core.util.switchresiduetypeset: [ WARNING ] When switching to a fa_standard ResidueTypeSet:  Pose already contains fa_standard ResidueTypes.
protocols.DsspMover: LHHHHLLLLLHHHHHHHHHHHHLLHHHHHHHHHHHHHLLLLLLHHHHHHHHHHLHHHHHHHHHHHHHHHHHLLLLLHHHHHHHHLLLHHHHHHHHHHHHHHHHHHHLHHHHHHHHHHHHHHHHLLLLLLLLLLLLLLLLLLHHHHHHHHHHHLLLLLLLLELLLLHHHHHHHHHHHHHHHHHHHHLHHHHHHHHHHHHHHHHHHHHHHHHHHHHHLEELLEEHHHHHHHHHLLLLLLLLLHHHHHHHHHHHHHHLLLLLLHHHHHHHHLLLLLLLHHHHHHHHHHHHHHHHLHHHHLHHHHHHHHHLHHHHHHHHHHLLLHHHHHHHHHHHHHLLLHHHHHHHHHHHHHHHHHHHHHHHHLLHHHHHHHLLLLLHHHHHHHHHHHHHHHHHHHHHELLHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHLLLLLLLLHHHHHHHHHHHHHHHHHHHHHHHHHLLHHHHHLLLHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHLLLLLLLL
protocols.membrane.MPLipidAccessibility: nbeta: 4 nmem: 173 beta: 0.0231214
protocols.membrane.MPLipidAccessibility: -15.351 15.351 10.234 6 10
core.membrane.hull: concave shell
core.membrane.hull: convex hull

AN INTERNAL ERROR HAS OCCURED. PLEASE SEE THE CONTENTS OF ROSETTA_CRASH.log FOR DETAILS.
zsh: segmentation fault   @score.options -in:file:s 2A65_full.pdb 3TT3_full.pdb -out:pdb

**Comment by biocatchen** [comment]
[ROSETTA_CRASH.log](https://github.com/user-attachments/files/17007099/ROSETTA_CRASH.log)

**Reply by biocatchen** [reply]
forgot to add the crash.log file in the previous post

**Reply by biocatchen** [reply]
No response yet

**Comment by roccomoretti** [comment]
So `score_jd2.static.macosclangrelease` and `score_jd2.linuxgccrelease` should be entirely equivalent. It's really the first part of the executable name (the `score_jd2`) that matters. The rest is just details for how the program was compiled. This may be of concern in certain situations (e.g. if you want to run under MPI, you'll need the mpi version), but for general use case it usually doesn't. -- In particular, the `static` simply refers to how the program executable is put together and shouldn't change behavior.

Regarding you crash, from what I can tell it's related to the fact that your membrane definition is such that no residues are considered to be within the membrane. It then falls over on itself during the convex hull calculation.

I'm not quite sure why the membrane definition is giving you issues. One potential cause might be a PDB numbering/pose numbering mismatch. The residue numbers in the membrane span file are _pose_ numbers (you can tell as they don't have any chain information associated with them). Pose numbering is a sequential numbering starting from 1. This is in contrast to the PDB numbering, which for the 2A65 downloaded from the PDB has the first residue starting at number 5. I'm not entirely sure why such a shift results in a membrane span definition which puts things outside the membrane, but changing the spanfile (though not necessarily correctly) does seem to fix the issue in my local test.
